# Um pouco de estatística

In [2]:
import pandas as pd

url = 'http://personal.tcu.edu/kylewalker/data/colleges.csv'
data = pd.read_csv(url, encoding = 'latin_1')
data.shape

(7804, 122)

Observamos que nosso conjunto de dados possui mais de 7800 linhas e 122 colunas! Neste ponto, embora você ainda não saiba quais insights estão contidos nos dados, já deve ter uma ideia das perguntas de pesquisa que lhe interessam. Como essas perguntas provavelmente não exigem todas as 122 colunas, é seguro restringir o conjunto de dados apenas às colunas necessárias para sua análise. Nossa principal pergunta de pesquisa é:

**Como a proporção de estudantes alunos de graduação com 25 anos ou mais varia entre faculdades e universidades que possuem bacharelado?**

Para responder a essa pergunta, precisamos identificar algumas colunas essenciais e filtrar os dados de acordo. As colunas que manteremos são as seguintes:

- **INSTNM**: Nome da instituição;
- **STABBR**: Estado onde a instituição está localizada;
- **PREDDEG**: Tipo principal de diploma concedido pela instituição (códigos: 1 para certificados, 2 para diplomas de associado, 3 para bacharelados e 4 para pós-graduação);
- **CONTROL**: Propriedade da instituição (códigos: 1 para pública sem fins lucrativos, 2 para privada sem fins lucrativos, e 3 para privada com fins lucrativos);
- **UGDS**: Número de alunos de graduação matriculados na instituição;
- **UG25abv**: Percentual de alunos de graduação com 25 anos ou mais na instituição.


In [ ]:
cols = ['INSTNM', 'STABBR', 'PREDDEG', 'CONTROL', 'UGDS', 'UG25abv']
df = data.filter(cols)

df.columns = [col.lower() for col in df.columns]

df.rename(columns={'stabbr': 'state', 'instnm': 'institution'}, inplace=True)

dfNotNull = df.dropna()